reconcile/utils/terrascript_client.py

```diff
@@ -4451,3 +4465,190 @@ def populate_tf_resource_route53_zone(self, resource, namespace_info):
         )
 
         self.add_resources(account, tf_resources)
+
+    def populate_tf_resource_cognito(self, resource, namespace_info):
+        account, identifier, common_values, output_prefix, \
+            output_resource_name, annotations = \
+            self.init_values(resource, namespace_info)
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info, output_prefix,
+                                output_resource_name, annotations)
+
+        # FIXME
+        # These should be sourced from ai provider values
+        sms_role_ext_id = "foobar"
+        organization_name = "rosa-fedramp"
+        cognito_user_pool_domain = "cognito.fedramp.devshift.net"
+
+        sms_role_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "sts:AssumeRole",
+                    ],
+                    "Principal": {
+                        "Service": "cognito-idp.amazonaws.com",
+                    },
+                    "Condition": {
+                        "StringEquals": {
+                            "sts:ExternalId": sms_role_ext_id
+                        }
+                    }
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "sns:Publish",
+                    ],
+                    "Resource": "*",
+                }
+            ]
+        }
+
+        sms_iam_role_resource = aws_iam_role(
+            "sms_role",
+            name=f'{organization_name}-SMS',
+            description="role for applicant cognito, send sms",
+            assume_role_policy=sms_role_policy,
+            force_detach_policies=False,
+            max_session_duration=3600,
+            path="/service-role/",
+        )
+        tf_resources.append(sms_iam_role_resource)
+
+        # https://registry.terraform.io/providers/hashicorp/aws/3.60.0/docs/resources/cognito_identity_provider
+
+        # What is the difference between values and self.get_values?
+        # How do they relate to default files vs supplied parameters?
```

Comment:
In app-interface namespace files, `defaults` and `parameter_group` both are expected to use a app-interface YML filepath as a value. Therefore I would expect to see the `self.get_values()` function to be used at any point in this code file when we need to read the data from those YML files, and I would expect to see `values.get()` or `common_values.get()` to be used if we're just checking for the presence of the key in the namespace file.

Example: https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/advisor/namespaces/stage-advisor-stage.yml#L71-76